### PR TITLE
added a string resources class UiText.kt in core:common this function…

### PR DESCRIPTION
**Pull request by ALI-M0vS5a**
… can be used in the entire project in order to have access to string resources also i have refactored the core:module by adding a manifest and the access to the core:common in core:common in the NewsResourceType.kt i have used the UiText.kt to change the description to stringResources

### CODE REVIEW

**Comment by dturner** [comment]
Hey thanks for submitting this. It looks like a great solution.

**Comment by dturner** [comment]
Please run `./gradlew --init-script gradle/init.gradle.kts spotlessApply` to fix the formatting issues.

**Comment by ALI-M0vS5a** [comment]
hello.
should i remove the org.gradle.java.home=C\:\\Program Files\\Java\\jdk-11 from gradle.properties after running this 
./gradlew --init-script gradle/init.gradle.kts spotlessApply successfully?

**Review by alexvanyo — COMMENTED** [review]
I'm a bit confused by this PR, since it looks like `asString` and `unknownError` are both unused, and instances of `DynamicString` are never created.

Is the goal to have a more robust string mechanism for when we do need those methods?

File: core/common/src/main/java/com/google/samples/apps/nowinandroid/core/uitext/UiText.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+package com.google.samples.apps.nowinandroid.core.uitext
+
+import android.content.Context
+import androidx.annotation.StringRes
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.res.stringResource
+import com.google.samples.apps.nowinandroid.core.common.R
+
+sealed class UiText {
```
Comment (mmoczkowski) [inline core/common/src/main/java/com/google/samples/apps/nowinandroid/core/uitext/UiText.kt:25]: This wrapper class is unnecessary noise and complexity imo

File: core/model/src/main/java/com/google/samples/apps/nowinandroid/core/model/data/NewsResourceType.kt [hunk 2]
```diff
@@ -16,59 +16,61 @@
...
+
 /**
  * Type for [NewsResource]
  */
 enum class NewsResourceType(
     val serializedName: String,
     val displayText: String,
-    // TODO: descriptions should probably be string resources
-    val description: String
+    val description: UiText
```
Comment (mmoczkowski) [inline core/model/src/main/java/com/google/samples/apps/nowinandroid/core/model/data/NewsResourceType.kt:28]: I think it's more beneficial to simply store a resource id here

**Review by mmoczkowski — CHANGES_REQUESTED** [review]
Remove the wrapper class and make description field a resource id instead of a string

**Comment by mmoczkowski** [comment]
Closing due to inactivity. If you still wish to submit this change, please create a new PR.

**Pull request closed without merging**